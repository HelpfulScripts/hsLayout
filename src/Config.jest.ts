import { Config }   from './';
import * as layout  from './';
import m from "mithril";

// const fs = node.fs;

const myConfig = {
    root: { Layout: {
        css: '.my-example',
        rows:    ['30px', 'fill', '10px'],
        content: ['Header', 'Body', 'Footer']
    }},
    Header: { Layout:{
        columns: ['200px', 'fill'],
        content: ['LeftHead', 'MainHead']
    }},
    Body: { Layout:{
        columns: ['200px', 'fill'], 
        content: ['LeftNav', 'MainNav']
    }},
    Footer: { Layout: {
        css: '.hs_footer',
        content: ['(c) Helpful ; Scripts']
    }},
    LeftHead: "left head",
    MainHead: "main head",
    LeftNav:  "left nav",
    MainNav:  "main nav"
};


// Polyfill DOM env for mithril
window = require("mithril/test-utils/browserMock.js")();
document = window.document;


describe('Config', () => {
    const root = document.createElement("div");
    beforeAll(() => {
        m.mount(root, {view: () => m(Config, {source:myConfig, context:[layout]})});
    });
    
    test('Config should match snapshot', () => {
        expect(root).toMatchSnapshot();
    });
});

