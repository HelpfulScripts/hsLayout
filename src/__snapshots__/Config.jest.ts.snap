// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Config should match snapshot 1`] = `
<div>
  <div
    class="hs-layout  hs-row-layout  my-example"
  >
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; top: 0px; height: 30px;"
    >
      <div
        class="hs-layout  hs-column-layout "
      >
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 0px; width: 200px;"
        >
          <div
            class="hs-leaf"
          >
            left head
          </div>
        </div>
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 200px; right: 0px; width: auto;"
        >
          <div
            class="hs-leaf"
          >
            main head
          </div>
        </div>
      </div>
    </div>
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; top: 30px; bottom: 10px; height: auto;"
    >
      <div
        class="hs-layout  hs-column-layout "
      >
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 0px; width: 200px;"
        >
          <div
            class="hs-leaf"
          >
            left nav
          </div>
        </div>
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 200px; right: 0px; width: auto;"
        >
          <div
            class="hs-leaf"
          >
            main nav
          </div>
        </div>
      </div>
    </div>
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; bottom: 0px; height: 10px;"
    >
      <div
        class="hs-layout   hs-site-footer"
      >
        <div
          class="hs-layout  "
        >
          <div
            class="hs-leaf"
          >
            (c) Helpful ; Scripts
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Config file Config from file should match snapshot 1`] = `
<div>
  <div
    class="hs-layout  hs-row-layout  my-example"
  >
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; top: 0px; height: 30px;"
    >
      <div
        class="hs-layout  hs-column-layout "
      >
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 0px; width: 200px;"
        >
          <div
            class="hs-leaf"
          >
            left head
          </div>
        </div>
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 200px; right: 0px; width: auto;"
        >
          <div
            class="hs-leaf"
          >
            main head
          </div>
        </div>
      </div>
    </div>
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; top: 30px; bottom: 10px; height: auto;"
    >
      <div
        class="hs-layout  hs-column-layout "
      >
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 0px; width: 200px;"
        >
          <div
            class="hs-leaf"
          >
            left nav
          </div>
        </div>
        <div
          class="hs-layout  "
          style="top: 0%; bottom: 0%; left: 200px; right: 0px; width: auto;"
        >
          <div
            class="hs-leaf"
          >
            main nav
          </div>
        </div>
      </div>
    </div>
    <div
      class="hs-layout  "
      style="left: 0%; right: 0%; bottom: 0px; height: 10px;"
    >
      <div
        class="hs-layout   hs-site-footer"
      >
        <div
          class="hs-layout  "
        >
          <div
            class="hs-leaf"
          >
            (c) Helpful ; Scripts
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
